src/geode/model/helpers/convert_to_mesh.cpp

```diff
@@ -878,25 +876,22 @@ namespace geode
         return new_convert_model_into_curve( brep );
     }
 
-    template < typename SurfaceType = SurfaceMesh3D >
-    std::tuple< std::unique_ptr< SurfaceType >, ModelToMeshMappings >
+    std::tuple< std::unique_ptr< SurfaceMesh3D >, ModelToMeshMappings >
         new_convert_brep_into_surface( const BRep& brep )
     {
-        return new_convert_model_into_surface< SurfaceType >( brep );
+        return new_convert_model_into_surface< SurfaceMesh3D >( brep );
     }
 
-    template < typename SurfaceType = SurfaceMesh2D >
-    std::tuple< std::unique_ptr< SurfaceType >, ModelToMeshMappings >
+    std::tuple< std::unique_ptr< SurfaceMesh2D >, ModelToMeshMappings >
         new_convert_section_into_surface( const Section& section )
     {
-        return new_convert_model_into_surface< SurfaceType >( section );
+        return new_convert_model_into_surface< SurfaceMesh2D >( section );
     }
 
-    template < typename SolidType = SolidMesh3D >
-    std::tuple< std::unique_ptr< SolidType >, ModelToMeshMappings >
+    std::tuple< std::unique_ptr< SolidMesh3D >, ModelToMeshMappings >
         new_convert_brep_into_solid( const BRep& brep )
     {
-        auto mesh = SolidType::create();
+        auto mesh = SolidMesh3D::create();
```

Comment:
expose this function and use it here https://github.com/Geode-solutions/OpenGeode/blob/3361d99ac4ce8d4412a7bbd1f44a7f2f662f7537/src/geode/mesh/helpers/detail/vertex_merger.cpp#L42